Source/Core/VideoBackends/Software/Tev.cpp

```diff
@@ -815,6 +815,9 @@ void Tev::Draw()
       fog = 1.0f - fog;
       fog = pow(2.0f, -8.0f * fog * fog);
       break;
+    case FogType::Linear:
+    case FogType::Off:
+      ASSERT(0);
```

Comment:
Most probable reason for the rendering failures would be this assert, which I added in order to see which games might use them since they weren’t actually handled.

But fifoci seems to be telling me that they are actually used. :)